dev-academy/develop-smart-contract/02-develop.md

```diff
@@ -2,61 +2,78 @@
-
-Here is the application logic we want:
-
-- Collected tokens in smart contract's balance are released to a target address after
-  the token amount exceeds a specified amount.
-- Contract accepts cw20 token that is predefined by the admin.
-
-:::warning
-We recommend deleting boilerplate code during implementation to help with copying and pasting code.
+In this section, we will build on top of the template that was generated [previously](01-intro.md), and develop a smart contract that will function as a To-Do List.
+:::tip Reminder
+The template contract is only being utilized to provide the initial structure. The various functions, messages and data structures that are defined in the template contract will need to be repurposed, replaced or removed in accordance with the needs of our new contract.
 :::
 
-:::info
-You can find the full version of this contract at [cw-contracts/cw20-pot](https://github.com/InterWasm/cw-contracts/tree/main/contracts)
-:::
+Here is the application logic:
+- The owner of the To-Do List contract can add new entries, update existing ones or delete them.  
+- The contract can be queried to return individual entries as well as a subset of the whole list. 
 
-## Config
+You can find the full version of this contract by clicking on the following link:
+* [cw-contracts/cw-to-do-list](https://github.com/InterWasm/cw-contracts/tree/main/contracts)
+## Contract Config & State
 
-`admin` information which determines who has pot creation address must be persisted on contract
-storage during each execution. We will save this information to the storage on `Instantiate` phase.
+Each instance of the To-Do List contract should have an `owner` with entry manipulation rights. In order to be able to check whether a function call is authorized or not, the address of the owner must be persisted on contract storage following the instantiation.
 
 ```rust
-// state.rs
+//state.rs
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct Config {
   pub owner: Addr,
-  pub cw20_addr: Addr
 }
 
 pub const CONFIG: Item<Config> = Item::new("config");
+//Item stores a single variable of a given type, identified by a string storage key.
 ```
+A To-Do List consists of entries, each of which is identified by a unique `id`. The `id` of the latest entry on the list is stored as `ENTRY_SEQ`, an Item that holds an integer value. The value stored in `ENTRY_SEQ` is incremented every time a new entry is appended to the list.
 
-`Item` is used since this data will be a singleton.
+The attributes `status` and `priority` are defined as enums to provide uniformity among different entries.
 
-## Instantiate
+The list of entries as a whole is persisted as the `LIST`, a Map that holds a collection of `id` to `Entry` pairs.
 
-### Msg
+```rust
+//state.rs
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Entry {
+    pub id: Uint64,
```

Comment:
We can simply use `u64` for keys in storage-plus, best to reflect that in the docs